src/ResourceManagement/AppService/DeploymentSlotBaseImpl.cs

```diff
@@ -0,0 +1,385 @@
+            return await base.SubmitAppSettingsAsync(Inner, cancellationToken);
+        }
+
+        ///GENMHASH:D9CEBC2EF6EA60751F2BFE78FE5E22B3:86C919F4CA67F8BA602AC70FD7CCC9E0
+        internal override async Task<Models.SiteInner> SubmitConnectionStringsAsync(SiteInner site, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (configurationSource == null || !IsInCreateMode)
+            {
+                return await base.SubmitConnectionStringsAsync(site, cancellationToken);
+            }
+
+            foreach (IConnectionString conn in (await configurationSource.GetConnectionStringsAsync(cancellationToken)).Values)
+            {
+                if (conn.Sticky)
+                {
+                    WithStickyConnectionString(conn.Name, conn.Value, conn.Type);
+                }
+                else
+                {
+                    WithConnectionString(conn.Name, conn.Value, conn.Type);
+                }
+            }
+            return await base.SubmitConnectionStringsAsync(Inner, cancellationToken);
+        }
+
+        ///GENMHASH:07FBC6D492A2E1E463B39D4D7FFC40E9:6015327ABEB713972CA126D7A0F3C232
+        internal override async Task<Models.SiteInner> CreateOrUpdateInnerAsync(SiteInner site, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await Manager.Inner.WebApps.CreateOrUpdateSlotAsync(ResourceGroupName, parent.Name, site, Name, cancellationToken: cancellationToken);
+        }
+
+        ///GENMHASH:21FDAEDB996672BE017C01C5DD8758D4:42826DB217BC1AEE5AAA977944F4318D
+        internal override async Task<Models.ConnectionStringDictionaryInner> UpdateConnectionStringsAsync(ConnectionStringDictionaryInner inner, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await Manager.Inner.WebApps.UpdateConnectionStringsSlotAsync(ResourceGroupName, parent.Name, inner, Name, cancellationToken);
+        }
+
+        ///GENMHASH:8E71F8927E941B28152FA821CDDF0634:FE3484EE3177CF2A5BBF1D4825D24686
+        internal override async Task<Models.SiteAuthSettingsInner> GetAuthenticationAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await Manager.Inner.WebApps.GetAuthSettingsSlotAsync(ResourceGroupName, parent.Name, Name, cancellationToken);
+        }
+
+        ///GENMHASH:4F0DD1E3F09332DAEE78A7163765E0EA:DC09185D93597A9B3912ED6CC825299E
+        public override async Task<Microsoft.Azure.Management.AppService.Fluent.IPublishingProfile> GetPublishingProfileAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Stream stream = await Manager.Inner.WebApps.ListPublishingProfileXmlWithSecretsSlotAsync(ResourceGroupName, parent.Name, new CsmPublishingProfileOptionsInner(), Name, cancellationToken);
+            StreamReader reader = new StreamReader(stream);
+            string xml = reader.ReadToEnd();
```

Comment:
You should dispose StreamReader. Please use using() statement.

Reply:
Even Stream has the Disposable implementation. So please add using over that one. 

---
In reply to: [150992462](https://github.com/Azure/azure-libraries-for-net/pull/91#discussion_r150992462) [](ancestors = 150992462)